test/e2e/tls/tls_test.go

```diff
@@ -18,14 +18,24 @@ import (
 var testRuntime *e2e.TestRuntime
 var pool *x509.CertPool
 
+var minTLSVersions = map[string]uint16{
+	"1.0": tls.VersionTLS10,
+	"1.1": tls.VersionTLS11,
+	"1.2": tls.VersionTLS12,
+	"1.3": tls.VersionTLS13,
+}
+
 // print error to stderr, exit 1
 func fatal(err interface{}) {
 	fmt.Fprintf(os.Stderr, "%s\n", err)
 	os.Exit(1)
 }
 
 func TestMain(m *testing.M) {
+	minTLSVersion := flag.String("--min-TLS-Version", "1.2", "minimum TLS Version")
+	TLSVersion := minTLSVersions[*minTLSVersion]
```

Comment:
Hrmm I don't quite understand why we're going through flag here. It seems quite indirect, and if this is why we need to mess with `os.Args` below, I'd rather keep looking for a different solution. (If this style is consistent with the other e2e tests, which I haven't check yet, please ignore me 😅)

I suppose we could set `testServerParams.MinTLSVersion` directly. If we need a different value for the other test below, could we change that field...?

Reply:
@srenatus all the tests run through same opa server instance. 
We have two set of testcases -
1. default tls version set for server
2. TLS1.3 set for server as implemented [here](https://github.com/open-policy-agent/opa/pull/3517/commits/2997f32031dfe1718d942b7747e6d8c44b49ed15#diff-166573a01b16eed2400e03bfa4f6acf455aacad1f7827856250008684ea3408cR129) )
Hence to pass minTLSVersion to testServerParams through TestMain at runtime, I had to use os.Args, else I would have had to create two different servers for these testcases